cmd/dptp-pools-cm/main.go

```diff
@@ -54,13 +51,11 @@ type poolsPullSecretProviderOptions struct {
 }
 
 func newOpts() (*options, error) {
-	opts := &options{}
+	opts := &options{kubernetesOptions: flagutil.KubernetesOptions{NOInClusterConfigDefault: true}}
```

Comment:
The logic here seems wrong. What does this tool expect? Why do we allow *either* a `hive` context _or_ in-cluster config? What's the utility of this? Can we collapse to using just one? If not, why not configure the options here to load in-cluster, then in the code below simply use it when no `hive` config is passed? No need to load twice.

Reply:
NOInClusterConfigDefault skips this function at upstream.

https://github.com/kubernetes/test-infra/blob/665b28a81b9e9c97420ca03b5937a518938527f0/prow/kube/config.go#L51

It generates 2 more keys in the kubeconfig map: "" and "default".
The kubeconfig for "" is the inClusterConfig and "default" is the cluster Prow runs.
I feel those things are only for Prow and our tools should not include the logic.

If `opts := &options{kubernetesOptions: flagutil.KubernetesOptions{}}`, then I have to those 2 keys for build-clusters.

Reply:
You check later to see if the hive context exists, and if not, load the in-cluster config anyway. Why?

Reply:
Copied code from dptp-cm to support both the env. var and inClusterConfig. Actually, we can just use the inClusterConfig for this tool since it works only with `hive` and we use the SA for the pod.

https://github.com/openshift/release/blob/b809b4e30b517d13a1a6db73e1a989900636ac8a/clusters/hive/dptp-pools-cm.yaml#L114